tests/avian.rs

```diff
@@ -1,106 +1,459 @@
-    ));
-
-    // Cube
-    commands.spawn((
-        Transform::from_xyz(-5.0, 0.8, -5.0),
-        Collider::cuboid(1.25, 1.25, 1.25),
-        NavMeshAffector,
-    ));
-
-    // Tall Cube
-    commands.spawn((
-        Transform::from_xyz(-0.179, 18.419, -27.744).with_scale(Vec3::new(15.0, 15.0, 15.0)),
-        Collider::cuboid(1.25, 1.25, 1.25),
-        NavMeshAffector,
-    ));
-
-    // Thin wall
-    commands.spawn((
-        Transform::from_xyz(-3.0, 0.8, 5.0).with_scale(Vec3::new(50.0, 15.0, 1.0)),
-        Collider::cuboid(0.05, 0.05, 0.05),
-        NavMeshAffector,
-    ));
+#[test]
+fn test_simple_navigation() {
+    let path = App::setup_test_world().setup_world().run_pathfinding();
+
+    if let Err(error) = path {
+        panic!("Pathfinding failed: {error:?}");
+    }
+}
+
+#[test]
+fn nav_mesh_is_cleared() {
+    let path = App::setup_test_world()
+        .setup_world()
+        .clear_world()
+        .run_pathfinding();
+
+    assert!(path.is_err());
 }
 
-fn setup_app(app: &mut App) {
-    app.add_plugins((
-        MinimalPlugins,
-        TransformPlugin,
-        OxidizedNavigationPlugin::<Collider>::new(NavMeshSettings {
-            cell_width: 0.25,
-            cell_height: 0.1,
-            tile_width: NonZeroU16::new(100).unwrap(),
-            world_half_extents: 250.0,
-            world_bottom_bound: -100.0,
-            max_traversable_slope_radians: (40.0_f32 - 0.1).to_radians(),
-            walkable_height: 20,
-            walkable_radius: 1,
-            step_height: 3,
-            min_region_area: 100,
-            max_region_area_to_merge_into: 500,
-            max_contour_simplification_error: 1.1,
-            max_edge_length: 80,
-            max_tile_generation_tasks: NonZeroU16::new(8), // Github Actions are limited to 7 GB.
-            experimental_detail_mesh_generation: None,
-        }),
-        PhysicsPlugins::default(),
-        HierarchyPlugin,
-    ));
+#[test]
+fn nav_mesh_is_deterministic() {
+    let mut app = App::setup_test_world();
+
+    let nav_mesh_one = app.setup_world().get_nav_mesh();
+    app.clear_world();
+    let nav_mesh_two = app.setup_world().get_nav_mesh();
+
+    assert_nav_mesh_equal(nav_mesh_one, nav_mesh_two);
 }
 
-fn wait_for_generation_to_finish(app: &mut App) {
-    loop {
-        app.update();
+#[test]
+fn compound_colliders_create_same_navmesh_as_individual_colliders_with_no_thin_wall() {
+    let mut app = App::setup_test_world();
+
+    let nav_mesh_one = app.setup_no_thin_wall().get_nav_mesh();
+    app.clear_world();
+    let nav_mesh_two = app.setup_compound_no_thin_wall().get_nav_mesh();
 
-        if app.world().resource::<ActiveGenerationTasks>().is_empty() {
-            break;
-        } else if app.world().resource::<Time>().elapsed() >= TIMEOUT_DURATION {
-            panic!("Generation timed out.");
+    assert_nav_mesh_equal(nav_mesh_one, nav_mesh_two);
+}
+
+#[test]
+fn compound_colliders_create_same_navmesh_as_individual_colliders() {
```

Comment:
This test fails. If you think that there's no actual bug involved, we can delete this one.

Reply:
The analogous test for non-compounds works every time, so I'm not sure the ECS is to blame

Reply:
Yeah, I double checked, the ECS doesn't iterate backwards. Probably remembering another library.

As long as the compound colliders are themselves deterministic I think it's fine. (i.e recreating from compound collider is the same, individual being the same as compound is not necessary)

Reply:
I'll add a test

Reply:
Works! https://github.com/TheGrimsey/oxidized_navigation/pull/42/commits/d3471ab817bf6303c816855c6cdf65429b12226d